internal/controllers/cluster/cluster_controller.go

```diff
@@ -147,6 +140,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 		}
 	}()
 
+	// Return early if the object or Cluster is paused.
+	if annotations.IsPaused(cluster, cluster) {
+		log.Info("Reconciliation is paused for this object")
```

Comment:
TODO: this is now always logged, we should only log it once, when we change the condition!